File: quiz/src/main/java/com/example/quizbackend/demo/domain/Question.java
package com.example.quizbackend.demo.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import java.util.List;

@Table(uniqueConstraints={@UniqueConstraint(columnNames ={"label","theme"})})
@Entity
@JsonIgnoreProperties(ignoreUnknown = true)
@Data
public class Question {
    @Id
    @SequenceGenerator(name = "question", sequenceName = "question_seq")
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "question")
    private Long id;
    private String label;
    private String theme;
    @OneToMany
    @NotEmpty
    private List<Answer> answerList;
    @OneToOne
    @JsonIgnore
    private Answer rightAnswer;
    @Transient
    private String proposition;

}

File: quizfront/src/app/domain/Question.ts
import {Answer} from "./Answer";

export class Question {
  private label:string;
  private theme:string;
  private answerList: Answer[];
  private rightAnswer: Answer;


  get getlabel(): string {
    return this.label;
  }

  set setlabel(value: string) {
    this.label = value;
  }

  get gettheme(): string {
    return this.theme;
  }

  set gettheme(value: string) {
    this.theme = value;
  }

  get getanswerList(): Answer[] {
    return this.answerList;
  }

  set getanswerList(value: Answer[]) {
    this.answerList = value;
  }

  get getrightAnswer(): Answer {
    return this.rightAnswer;
  }

  set getrightAnswer(value: Answer) {
    this.rightAnswer = value;
  }
}
